src/main/python/analyzers/project_analyzer.py

```diff
@@ -1,77 +1,126 @@
+import pandas as pd
 import analyzers.sentiment_analyzer as sa
+import analyzers.embedded_analyzer as emb
 import analyzers.emotion_analyzer as ea
-import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LinearRegression
+from sklearn.neural_network import MLPRegressor
+from sklearn.ensemble import RandomForestRegressor
 
 
 class TweetsAnalyzer:
     def __init__(self, tweetsAndStatsDF):
         self.tweetsAndStatsDF = tweetsAndStatsDF
 
-    def getSentimentAnalysis(self, tweetsList: list.__class__, regression=True):
-        resultsDF = sa.getSentimentAnalysis(tweetsList)
-        print("Results\n", resultsDF.index)
-        mergedDF = pd.merge(
-            left=self.tweetsAndStatsDF.copy(),
-            right=resultsDF,
-            left_index=True,
-            right_index=True
-        )
-        print("Merged\n", mergedDF)
-        # if regression:
-        #     resultsDictOrDF = __performRegressionAnalysis(resultsDF, featureColumnIndexes))
-        # else:
-        #     resultsDictOrDF = __performOtherAnalysis(resultsDF, featureColumnIndexes)
-        # return resultsDictOrDF
 
-    def getEmotionAnalysis(self, regression=True):
+    def getSentimentAnalysis(self, regressionMethod='1'):
+        resultsDF = sa.getSentimentAnalysis(self.tweetsAndStatsDF.copy())
+        featureColumnIndexes = ['positive', 'neutral', 'ngeative']
+        # Combine resultsDF with self.tweetsAndStats.copy()
+        if regressionMethod == '1':
+            resultsDF = self.__performRegressionAnalysis(resultsDF, featureColumnIndexes)
+        elif regressionMethod == '2':
+            resultsDF = self.__performMLPAnalysis(resultsDF, featureColumnIndexes)
+        else:
+            resultsDF = self.__performRandomForestAnalysis(resultsDF, featureColumnIndexes)
+        return resultsDF
+
+    def getEmotionAnalysis(self, regressionMethod='1'):
         resultsDF = ea.getEmotionAnalysis(self.tweetsAndStatsDF.copy())
+        featureColumnIndexes = ['fear', 'anger', 'trust', 'surprise', 'sadness', 'disgust', 'joy', 'anticipation']
         # Combine resultsDF with self.tweetsAndStats.copy()
-        # if regression:
-        #     resultsDictOrDF = __performRegressionAnalysis(resultsDF, featureColumnIndexes))
-        # else:
-        #     resultsDictOrDF = __performOtherAnalysis(resultsDF, featureColumnIndexes)
-        # return resultsDictOrDF
+        if regressionMethod == '1':
+            resultsDF = self.__performRegressionAnalysis(resultsDF, featureColumnIndexes)
```

Comment:
You don't have to change this, but usually when you send a df, you want to send a copy. It could be done in parts that I worked on too.